### PR TITLE
new eslint-plugin-lib

### PR DESCRIPTION
**Pull request by unicop**
# motivation

Have a single eslint-plugin under our name that will make it easy to customize different environments easily

### CODE REVIEW

File: packages/eslint-plugin/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "tests/fixtures/foo.ts",
+    "tests/fixtures/unstrict/file.ts",
+    "tests/fixtures/file.ts",
+    "tests/fixtures/class.ts"
+  ],
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
+      "parserOptions": {
+        "ecmaVersion": 2020,
```
Comment (neriyarden) [inline packages/eslint-plugin/.eslintrc.json:15]: isn't `{ "env": { "es6": true } }` needed too?

File: packages/eslint-plugin/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        // 'node-ts-only': configs['node-ts-only'], //'' as never,
+        'node-js-with-ts': '' as never,
+        // front
+        'jest-with-tsx-react': '' as never,
+        'jest-with-jsx-react': '' as never,
+        'jest-with-jsx-and-tsx-react': '' as never,
+        'react-tsx': '' as never,
+        'react-ts-only': '' as never,
+        'react-js-with-ts': '' as never,
+    },
```
Comment (neriyarden) [inline packages/eslint-plugin/src/index.ts:35]: react-js-only?

**Review by neriyarden — APPROVED** [review]
💪🏻

File: packages/eslint-plugin/.eslintrc.json (hunk 1 above)
Comment (unicop) [reply packages/eslint-plugin/.eslintrc.json:15]: we want to isolate things, I want to focus here on jest, and not which modules the user uses

Comment (unicop) [reply packages/eslint-plugin/.eslintrc.json:15]: ohh it is on our own library, I can add it :)

File: packages/eslint-plugin/src/index.ts (hunk 2 above)
Comment (unicop) [reply packages/eslint-plugin/src/index.ts:35]: yes we will haver that one, for now it is not that matter

**Comment by unicop** [comment]
@neriyarden I also added support for everything related to modules (esm,commonjs), I will merge it and we will combine everything soon

**Pull request merged**